src/main/java/com/meta/cp4m/S3PreProcessor.java

```diff
@@ -29,24 +33,26 @@ public class S3PreProcessor<T extends Message> implements PreProcessor<T> {
   private final String region;
   private final String bucket;
   private final @Nullable String textMessageAddition;
-  private final StaticCredentialsProvider credentialsProvider;
+  private @Nullable StaticCredentialsProvider staticCredentialsProvider;
```

Comment:
should be any reason to remove the `final` here

Reply:
Moved the 2nd assignment to variable in `else` condition which allows on-time assignment, in turn allowing `final`